### ISSUE TITLE
Crash in the threadExample project.

### ISSUE THREAD

**Issue by whataa** [realm/realm-java#3244]
```java
 Caused by: java.lang.IllegalStateException: Changing Realm data can only be done from inside a transaction.
                                                                               at io.realm.internal.Table.throwImmutable(Table.java:1375)
                                                                               at io.realm.internal.Table.checkImmutable(Table.java:1030)
                                                                               at io.realm.internal.Table.clear(Table.java:320)
                                                                               at io.realm.Realm.delete(Realm.java:1265)
                                                                               at io.realm.examples.threads.PassingObjectsFragment.onDestroy(PassingObjectsFragment.java:116)
```
```java
@Override
    public void onDestroy() {
        super.onDestroy();
        // Clear out all Person instances.
        realm.delete(Person.class);
        realm.close();
    }
```
According to the document, `delete` need to put in a transaction.

**Comment by cmelchior** [comment]
Well, that is embarrassing. We should get that fixed asap. Thanks for reporting it.

**Comment by cmelchior** [comment]
Fixed by #3245
